fastddsspy_participants/src/cpp/model/DataStreamer.cpp

```diff
@@ -73,18 +87,27 @@ void DataStreamer::add_data(
 {
     std::shared_lock<std::shared_timed_mutex> _(mutex_);
 
-    if (callback_ && topic == activated_topic_)
+    if (activated_)
     {
-        auto it = types_discovered_.find(topic.type_name);
-        if (it == types_discovered_.end())
+        // If all activated, add it only if schema is available, otherwise skip
+        if (activated_all_ && !is_topic_type_discovered_nts_(topic))
         {
-            // The topic that is supposed to be activated has no associated type. This should not happen
-            utils::tsnh(STR_ENTRY
-                    << "Topic <" << topic << "> must not be activated if its type is not registered.");
+            return;
         }
 
-        // TODO: make map search more safe
-        (*callback_)(topic, it->second, data);
+        if (activated_all_ || activated_topic_ == topic)
```

Comment:
why do you need activate_all_?

Reply:
It says whether all topics must be printed, or if only one of them (the on in activated_topic_) must